README.md

```diff
@@ -8,20 +8,57 @@ Beside, Cyclone DDS is developed completely in the open and is undergoing the ac
 # Getting Started
 ## Building Cyclone DDS
 
-In order to build cyclone DDS you need to have installed on your host [cmake](https://cmake.org/download/) **v3.6.0** or higher, the [Java 8 JDK](http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html) or simply the [Java 8 RE](http://www.oracle.com/technetwork/java/javase/downloads/server-jre8-downloads-2133154.html), and [Apache Maven 3.5.x or higher](http://maven.apache.org/download.cgi).
+In order use Eclipse Cyclone DDS you need to have installed on your host [cmake](https://cmake.org/download/) **v3.6.0** or higher, [git](https://git-scm.com/),  the [Java 8 JDK](http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html) or simply the [Java 8 RE](http://www.oracle.com/technetwork/java/javase/downloads/server-jre8-downloads-2133154.html), and [Apache Maven 3.5.x or higher](http://maven.apache.org/download.cgi).
 
-Assuming that **git** is also available on your machine then, simply do:
+To obtain Eclipse Cyclone DDS, do
 
     $ git clone https://github.com/eclipse/cyclonedds.git 
     $ cd cyclonedds
     $ mkdir build
+
+Depending on whether you want to develop applications using Eclipse Cyclone DDS or contribute to Eclipse Cyclone DDS you can follow different procedures
+
+### For application developers
+
+To build and install the required libraries needed to develop your own applications that use Eclipe Cyclone DDS
+
+On Linux:
+
     $ cd build
-    $ cmake ../src
-    $ make
-    $ make install
+    $ cmake -DCMAKE_BUILD_TYPE=<build-type> -DCMAKE_INSTALL_PREFIX=<install-location> ../src
+    $ cmake --build . --target install
+
+On Windows:
+
+    $ cd build
+    $ cmake -DCMAKE_BUILD_TYPE=<build-type> -G "<generator-name>" -DCMAKE_INSTALL_PREFIX=<install-location> ../src
```

Comment:
Perhaps it would be better to give a sensible suggestion for the generator on Windows? Like:

```suggestion
    $ cmake -DCMAKE_BUILD_TYPE=<build-type> -G "Visual Studio 15 2017 Win64" -DCMAKE_INSTALL_PREFIX=<install-location> ../src
```

or some other <generator-name> following the -G option if another version of Visual Studio or a 32-bit build is desired.

Reply:
@eboasson  Since you are asking my opinion ... I can image that you would like to have a concrete example of a cmake install command and thus want a concrete value for the generator name. But then I would also expect a concrete value for the build-type. It looks a bit arbitrary to me provide a concrete value for the generator name, but not for the build-type.

Reply:
Fair point — one approach for that is to remove the CMAKE_BUILD_TYPE setting from the example, and mention the option and the fact that it defaults to ``RelWithDebInfo``.